fiftyone/core/utils.py

```diff
@@ -1901,6 +1902,16 @@ def to_slug(name):
 
 _T = t.TypeVar("_T")
 
+sync_task_executor = None
```

Comment:
One improvement would be to handle cleanup of the singleton when not needed.

In generally we want 1 or 0 thread pools. But we don't need an idle pool unless it is being used. I believe this can be accomplished using `with` instead of just a function call.

That being said we could add that later I think.